### DISCUSSION TITLE
Cannot create simple file with puppet locally or at client node

### DISCUSSION THREAD

**Discussion by maworsham** [puppetlabs/community#62] [Troubleshooting] [answered]
I have Puppet v8.5.1 running on Ubuntu 22.04. I have a puppetmaster and puppet agent node communicating with each other. Now I created a simple "helloworld" under the production modules path (/etc/puppetlabs/code/environments/production/modules/helloworld/) and a manifests directory as well. Inside said manifests directory, I have this simple init.pp file:
```
class helloworld {
  file { '/tmp/foobar.txt':
    ensure  => file,
    mode    => '0644',
    content => 'Hello World \n';
  }
}
```

Now when I attempt to run the following "puppet apply" command from the puppetmaster, it shows only the following and there is no file being created in the /tmp folder:
```
root@sandbox-puppetmaster:~# puppet apply /etc/puppetlabs/code/environments/production/modules/helloworld/manifests/init.pp
Notice: Compiled catalog for sandbox-puppetmaster.ddc.local in environment production in 0.02 seconds
Notice: Applied catalog in 0.01 seconds
```

Over on the puppet node, I ran the command:
```
root@sandbox-node1:/tmp# puppet agent --test
Info: Using environment 'production'
Info: Retrieving pluginfacts
Info: Retrieving plugin
Notice: Requesting catalog from sandbox-puppetmaster.ddc.local:8140 (10.20.27.7)
Notice: Catalog compiled by sandbox-puppetmaster.ddc.local
Info: Caching catalog for sandbox-node1
Info: Applying configuration version '1709918930'
Notice: Applied catalog in 0.01 seconds
```
And no file shows up in /tmp either.

What is it I am doing wrong?

**Comment by bastelfreak** [comment, marked as answer]
You created a class, but you didn't apply/assign it to your node. `puppet apply /etc/puppetlabs/code/environments/production/modules/helloworld/manifests/init.pp -e 'include helloworld'` should work. Or remove the class statement:

```puppet
  file { '/tmp/foobar.txt':
    ensure  => file,
    mode    => '0644',
    content => 'Hello World \n';
  }
```

**Reply by binford2k** [reply]
@maworsham when explaining this to programmers, I use a metaphor that makes it pretty clear what happened. Imagine that you were writing Python, or Ruby, or Go, or whatever language you prefer. What you wrote in `init.pp` was like *defining* a function. But you never actually *invoked* it.

When you `include` it on a node like @bastelfreak showed ☝️,  you "invoke" it on that node and make it do something.

General practice for this is to create a parallel file in the `examples` directory (`/etc/puppetlabs/code/environments/production/modules/helloworld/examples/init.pp`) that looks like this:

``` puppet
# example usage of the helloworld class
include helloworld
```

and then `puppet apply` that example file.
